Microsoft.Toolkit.Uwp.Input.GazeInteraction/GazePointer.cs

```diff
@@ -0,0 +1,918 @@
+            SetElementStateDelay(_offScreenElement, PointerState.Dwell, _defaultDwell);
+
+            _maxHistoryTime = DEFAULT_MAX_HISTORY_DURATION;    // maintain about 3 seconds of history (in microseconds)
+            _gazeHistory = new List<GazeHistoryItem>();
+        }
+
+        private void InitializeGazeInputSource()
+        {
+            if (!_initialized)
+            {
+                if (_roots.Count != 0 && _devices.Count != 0)
+                {
+                    if (_gazeInputSource == null)
+                    {
+                        _gazeInputSource = GazeInputSourcePreview.GetForCurrentView();
+                    }
+
+                    if (_gazeInputSource != null)
+                    {
+                        _gazeInputSource.GazeEntered += OnGazeEntered;
+                        _gazeInputSource.GazeMoved += OnGazeMoved;
+                        _gazeInputSource.GazeExited += OnGazeExited;
+
+                        _initialized = true;
+                    }
+                }
+            }
+        }
+
+        private void DeinitializeGazeInputSource()
+        {
+            if (_initialized)
+            {
+                _initialized = false;
+
+                _gazeInputSource.GazeEntered -= OnGazeEntered;
+                _gazeInputSource.GazeMoved -= OnGazeMoved;
+                _gazeInputSource.GazeExited -= OnGazeExited;
+            }
+        }
+
+        private void ActivateGazeTargetItem(GazeTargetItem target)
+        {
+            if (_activeHitTargetTimes.IndexOf(target) == -1)
+            {
+                _activeHitTargetTimes.Add(target);
+
+                // calculate the time that the first DwellRepeat needs to be fired after. this will be updated every time a DwellRepeat is
+                // fired to keep track of when the next one is to be fired after that.
+                var nextStateTime = GetElementStateDelay(target.TargetElement, PointerState.Enter);
+
+                target.Reset(nextStateTime);
+            }
+        }
+
+        private GazeTargetItem GetHitTarget(Point gazePoint)
+        {
+            GazeTargetItem invokable = null;
+
+            if (!IsAlwaysActivated)
+            {
+                invokable = NonInvokeGazeTargetItem;
+            }
+
+            switch (Window.Current.CoreWindow.ActivationMode)
+            {
+                case CoreWindowActivationMode.ActivatedInForeground:
+                case CoreWindowActivationMode.ActivatedNotForeground:
+                    var elements = VisualTreeHelper.FindElementsInHostCoordinates(gazePoint, null, false);
+                    var element = elements.FirstOrDefault();
+
+                    invokable = null;
+
+                    if (element != null)
+                    {
+                        invokable = GazeTargetItem.GetOrCreate(element);
+
+                        while (element != null && !invokable.IsInvokable)
```

Comment:
Ah, good looks like we're using the TreeHelper here. Wondering if we could better optimize all the treehelper stuff in general. @Sergio0694 thoughts?

Reply:
Yeah, we can track improvements too in the issue you opened here: https://github.com/windows-toolkit/WindowsCommunityToolkit/issues/3487

Will resolve this specific use-case for now.

Reply:
Just realized we're not including the `Microsoft.Toolkit.Uwp.UI` project as a dependency here, so we don't access the tree helpers... We should decide if we want to add a dependency here and what that looks like? @azchohfi thoughts?